lpcompiler.pas

```
@@ -1228,18 +1229,32 @@ function TLapeCompiler.ParseMethodHeader(out Name: lpString; addToScope: Boolean
   Param: TLapeParameter;
   Token: EParserToken;
   Default: TLapeTree_ExprBase;
+  op: EOperator;
+  ltyp,rtyp:TLapeType;
 begin
   Pos := Tokenizer.DocPos;
-  isFunction := (Tokenizer.Tok = tk_kw_Function);
   Result := TLapeType_Method.Create(Self, nil, nil, '', @Pos);
-
+  Result.isOperator := (Tokenizer.Tok = tk_kw_Operator);
+  isFunction := (Tokenizer.Tok = tk_kw_Function);
+  
   try
-
-    if isNext([tk_Identifier, tk_sym_ParenthesisOpen], Token) and (Token = tk_Identifier) then
+    if (isNext([tk_Identifier, tk_sym_ParenthesisOpen], Token) and (Token = tk_Identifier)) or
+       (Result.isOperator and isNext(ParserToken_Operators, Token)) then
     begin
-      Name := Tokenizer.TokString;
-
-      if isNext([tk_sym_Dot, tk_sym_ParenthesisOpen], Token) and (Token = tk_sym_Dot) then
+      if not Result.isOperator then
+        Name := Tokenizer.TokString
+      else begin
+        for op in OverloadableOperators do
+          if ParserTokenToOperator(Tokenizer.Tok) = op then
+          begin
+            Name := '!op_'+op_name[op];
+            break;
+          end;
+        if Length(Name) = 0 then 
+          LapeExceptionFmt(lpeCannotOverloadOperator, [Tokenizer.TokString], Tokenizer.DocPos);
+      end;
+        
+      if isNext([tk_sym_Dot, tk_sym_ParenthesisOpen], Token) and (Token = tk_sym_Dot) and (not Result.isOperator) then
       begin
```

Maybe throw an exception if `(not Result.isOperator)`?